runtime/lua/vim/provider/python.lua

```diff
@@ -83,6 +83,10 @@ function M.detect_by_module(module)
     return vim.fn.exepath(vim.fn.expand(python_exe, true)), nil
   end
 
+  if vim.fn.executable('pynvim-python') == 1 then
+    return 'pynvim-python'
```

Comment:
`pynvim-python` isn't mentioned in provider.txt, I would expect some mention of it